docs/src/content/docs/book/gas-best-practices.mdx

```diff
@@ -76,6 +76,30 @@ contract Scrooge {
 }
 ```
 
+### Use `BasechainAddress` struct and related functions for runtime manipulations on addresses in the base workchain
+
+<Badge text="Available since Tact 1.6.3" variant="tip" size="medium"/><p/>
+
+Almost all contracts on TON are deployed on the basechain — a [workchain with ID 0][workchain-id]. The [`BasechainAddress{:tact}`](/ref/core-addresses#basechainaddress) [struct][struct] and related functions were introduced to allow optimal handling of basechain addresses at runtime.
+
+A frequent use-case for basechain addresses is checking message sender addresses for validity, i.e., if a contract needs to make sure the incoming message comes from a specific contract deployed in the basechain, it can use the [`StateInit.hasSameBasechainAddress(){:tact}`](/ref/core-addresses#stateinithassamebasechainaddress) extension function.
+
+```tact
+let init = initOf SomeContract();
+require(init.hasSameBasechainAddress(sender()), "Message must come from SomeContract");
+
+// Deriving the BasechainAddress from the StateInit for further use
+let addr: BasechainAddress = contractBasechainAddress(init);
+
+// Converting existing Address into the BasechainAddress
+let stdAddr: StdAddress = parseStdAddress(
+    address("EQCD39VS5jcptHL8vMjEXrzGaRcCVYto7HUn4bpAOg8xqB2N").asSlice()
+);
+let basechainAddr: BasechainAddress = newBasechainAddress(stdAddr.address);
```

Comment:
let's remove this part or move it to some other place in the docs, since it is non-essential to understanding this recipe